flash/core/integrations/labelstudio/data_source.py

```diff
@@ -0,0 +1,266 @@
+                                classes.add(sublabel)
+                                temp = {}
+                                temp["file_upload"] = task.get("file_upload")
+                                temp["data"] = task.get("data")
+                                if temp["file_upload"]:
+                                    temp["file_upload"] = os.path.join(data_folder, temp["file_upload"])
+                                else:
+                                    for key in temp["data"]:
+                                        p = temp["data"].get(key)
+                                    path = Path(p)
+                                    if path and data_folder:
+                                        temp["file_upload"] = os.path.join(data_folder, path.name)
+                                temp["label"] = sublabel
+                                temp["result"] = res.get("value")
+                                if annotation["ground_truth"]:
+                                    test_results.append(temp)
+                                elif not annotation["ground_truth"]:
+                                    results.append(temp)
+                        else:
+                            if isinstance(label, list):
+                                for item in label:
+                                    classes.add(item)
+                            else:
+                                classes.add(label)
+                            temp = {}
+                            temp["file_upload"] = task.get("file_upload")
+                            temp["data"] = task.get("data")
+                            if temp["file_upload"] and data_folder:
+                                temp["file_upload"] = os.path.join(data_folder, temp["file_upload"])
+                            else:
+                                for key in temp["data"]:
+                                    p = temp["data"].get(key)
+                                path = Path(p)
+                                if path and data_folder:
+                                    temp["file_upload"] = os.path.join(data_folder, path.name)
+                            temp["label"] = label
+                            temp["result"] = res.get("value")
+                            if annotation["ground_truth"]:
+                                test_results.append(temp)
+                            elif not annotation["ground_truth"]:
+                                results.append(temp)
+        return results, test_results, classes, data_types, tag_types
+
+
+class LabelStudioImageClassificationDataSource(LabelStudioDataSource):
+    """The ``LabelStudioImageDataSource`` expects the input to
+    :meth:`~flash.core.data.data_source.DataSource.load_data` to be a json export from label studio.
+    Export data should point to image files"""
+
+    def load_sample(self, sample: Mapping[str, Any] = None, dataset: Optional[Any] = None) -> Any:
```

Comment:
This doesn't seem tested.

Reply:
Tested with test_datasource_labelstudio_image